sdk/servicebus/azure-servicebus/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,1335 @@
+        else:
+            print("No messages available, waiting...")
+            time.sleep(1)
+```
+
+2. **Use async operations with proper cancellation:**
+```python
+import asyncio
+
+async def receive_with_cancellation(receiver):
+    try:
+        # Use asyncio timeout for better control
+        messages = await asyncio.wait_for(
+            receiver.receive_messages(max_message_count=10, max_wait_time=30),
+            timeout=35  # Slightly longer than max_wait_time
+        )
+        return messages
+    except asyncio.TimeoutError:
+        print("Receive operation timed out")
+        return []
+```
+
+### Messages not being received
+
+Messages might not be received due to various configuration or state issues.
+
+**Common causes and resolutions:**
+
+1. **Check entity state:**
+```python
+# Verify the queue/subscription exists and is active
+try:
+    # This will fail if entity doesn't exist
+    receiver = client.get_queue_receiver(queue_name)
+    messages = receiver.receive_messages(max_message_count=1, max_wait_time=5)
+    
+    if not messages:
+        print("No messages available - check if messages are being sent")
+    
+except MessagingEntityNotFoundError:
+    print("Queue/subscription does not exist")
+except MessagingEntityDisabledError:
+    print("Queue/subscription is disabled")
+```
+
+2. **Verify message filters (for subscriptions):**
+```python
+# For topic subscriptions, check if messages match subscription filters
+from azure.servicebus.management import ServiceBusAdministrationClient
+
+admin_client = ServiceBusAdministrationClient.from_connection_string(connection_string)
+
+# Check subscription rules
+rules = admin_client.list_rules(topic_name, subscription_name)
+for rule in rules:
+    print(f"Rule: {rule.name}, Filter: {rule.filter}")
+```
+
+3. **Check for competing consumers:**
+```python
+# Multiple receivers on the same queue will compete for messages
+# Ensure this is intended behavior or use topic/subscription pattern
+
+# For debugging, temporarily use peek to see if messages exist
+messages = receiver.peek_messages(max_message_count=10)
+print(f"Found {len(messages)} messages in queue without receiving them")
+```
+
+## Troubleshooting quota and capacity issues
+
+### Quota exceeded errors
+
+**ServiceBusQuotaExceededError resolution:**
+1. **For message count limits:** Receive and process messages to reduce queue/subscription size
+2. **For size limits:** Remove old messages or increase entity size limits
+3. **For connection limits:** Close unused connections or consider scaling to Premium tier
+
+### Entity not found errors
+
+**MessagingEntityNotFoundError resolution:**
+1. Verify the queue/topic/subscription name is spelled correctly
+2. Ensure the entity exists in the Service Bus namespace
+3. Check if the entity was deleted and needs to be recreated
+4. Verify you're connecting to the correct namespace
+
+## Threading and concurrency issues
```

Comment:
Move the Threading and concurrency issues section before the "Troubleshooting authentication issues" section.